src/main/java/decodex/Decodex.java

```diff
@@ -56,12 +56,12 @@ public static void main(String[] args) {
             String userInput = in.nextLine();
 
             try {
-                command = parser.parseCommand(userInput, dataManager, moduleManager, ui);
+                command = parser.parseCommand(userInput);
             } catch (ParserException err) {
                 printErrorMessage(err.getMessage());
                 continue;
             }
-            command.run();
+            command.run(dataManager, moduleManager, ui);
```

Comment:
Maybe we could add in a dummy string [] arguments, that way others can simply continue from this state?